src/core/string.rs

```diff
@@ -56,6 +57,15 @@ impl NgxStr {
         &self.0
     }
 
+    pub fn as_str(&self) -> &str {
+        // Safety: Converting the raw data to a string slice is unsafe, but as long
+        // as the lifetime of NgxStr is properly managed, this should be safe.
+        unsafe {
+            // let slice = std::slice::from_raw_parts(self.0.data, self.0.len);
+            std::str::from_utf8_unchecked(&self.0)
```

Comment:
Can we be guaranteed this is valid UTF-8? to_str is using the checked version now.